geos: do not set 3D on EWKB writer

otan opened the pull request:
geos: do not set 3D on EWKB writer

This seems to do weird things on certain systems if a GEOMETRYCOLLECTION
contains an EMPTY element.

The corollary of this is WKTs which are given 3D/4D "lose" their 3D/4D
component. I think this will have to be a docs limitation.

Since we don't support 3D/4D properly anyway, added a note to #53091.

Release justification: low risk update to new functionality

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53909)
<!-- Reviewable:end -->

otan committed:
geos: do not set 3D on EWKB writer

This seems to do weird things on certain systems if a GEOMETRYCOLLECTION
contains an EMPTY element.

The corollary of this is WKTs which are given 3D/4D "lose" their 3D/4D
component. I think this will have to be a docs limitation.

Release justification: low risk update to new functionality

Release note: None
